lms/static/js/student_account/views/FormView.js

```diff
@@ -226,7 +231,8 @@
 
                     if (!_.compact(this.errors).length) {
                         data = this.setExtraData(data);
-                        this.model.set(data);
+                        let newData = this.setUsaStateData(data);
```

Comment:
in case of implementation of `setUsaStateData` function don't need to assign a new value, just do `this.setUsaStateData(data);`